### PR TITLE
Prac 06 code review request

### PR DESCRIPTION
**Pull request by SallySheldon**
Please review my Prac 06 coding @dona-reano

### CODE REVIEW

File: prac_06/guitar.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+CP1404 Practical 06 - Do-from-scratch Exercises - Guitars!
+Define a Guitar class
+Estimated time to complete: 10 mins
+Actual time to complete: 20 mins
+"""
+
+import datetime  # Use datetime module to obtain the current year
+
+CURRENT_YEAR = datetime.date.today().year
+VINTAGE_AGE = 50
```
Comment (dona-reano) [inline prac_06/guitar.py:11]: Good use of constants and import datetime module :)

File: prac_06/guitar.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+class Guitar:
+    """Represent details about a guitar in a Guitar object."""
+
+    def __init__(self, name="", year=0, cost=0.0):
+        """Initialise a Guitar object.
+
+        name: string, name of guitar object
+        year: integer, year of manufacture
+        cost: float, cost in dollars and cents
+        """
```
Comment (dona-reano) [inline prac_06/guitar.py:23]: Good docsting detail 👍

File: prac_06/guitar.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        return f"{self.name} ({self.year}) : ${self.cost:,.2f}"
+
+    def __lt__(self, other):
+        """Sort Guitar objects by increasing order of year released."""
+        return self.year < other.year
+
+    def get_age(self):
+        """Return the age in years of a Guitar object using CURRENT_YEAR."""
+        age = CURRENT_YEAR - self.year
+        return age
```
Comment (dona-reano) [inline prac_06/guitar.py:39]: could just do this in one line by just returning CURRENT_YEAR - self.year 
otherwise good👍

File: prac_06/guitar.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    def is_vintage(self):
+        """Determine whether a Guitar object is vintage (at least VINTAGE_AGE)."""
+        age = self.get_age()
+        return age >= VINTAGE_AGE
```
Comment (dona-reano) [inline prac_06/guitar.py:44]: could do in one line as well 👍

File: prac_06/guitar.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+"""
+CP1404 Practical 06 - Do-from-scratch Exercises - Guitars!
+Define a Guitar class
+Estimated time to complete: 10 mins
+Actual time to complete: 20 mins
+"""
```
Comment (dona-reano) [inline prac_06/guitar.py:6]: Good top docstring 👍

File: prac_06/guitar_test.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+"""
+CP1404 Practical 06 - Do-from-scratch Exercises - Guitars!
+Program to test that Guitar class methods work as expected.
+Estimated time to complete: 15 mins
+Actual time to complete: 12 mins
+"""
```
Comment (dona-reano) [inline prac_06/guitar_test.py:6]: Good program detail 👍

File: prac_06/guitar_test.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+from prac_06.guitar import Guitar
+
+
+def run_tests():
+    """Program to test get_age() and is_vintage() methods in Guitar class definition."""
+    gibson = Guitar("Gibson L-5 CES", 1922, 16035.40)
+    another_guitar = Guitar("Another Guitar", 2013, 799.50)
```
Comment (dona-reano) [inline prac_06/guitar_test.py:14]: good name for variable 👍

File: prac_06/guitars.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    """Program to store and display details of a user's guitars."""
+    print("My guitars!")
+    guitars = []
+    name = input("Name: ")
+    # guitars.append(Guitar("Gibson L-5 CES", 1922, 16035.40))  # Test data
+    # guitars.append(Guitar("Line 6 JTV-59", 2010, 1512.9))  # Test data
+    while name != "":
+        year = int(input("Year: "))
+        cost = float(input("Cost: $"))
+        guitar = Guitar(name, year, cost)
```
Comment (dona-reano) [inline prac_06/guitars.py:21]: unclear variable name 'guitar'. does the guitar already exist in the list or does it need to be added as a new guitar? maybe the name should reflect what we are doing with it. something like guitar_to_add could be good to make it clearer 👍

File: prac_06/guitars.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        guitars.append(guitar)
+        print(f"{guitar} added.")
+        name = input("Name: ")
+
+    if guitars:  # First check whether the list has contents
+        guitars.sort()
+        # Calculate alignments for display formatting
+        max_number = len(str(len(guitars)))
+        max_name = max((len(guitar.name) for guitar in guitars))
+        max_cost = max((len(str(guitar.cost)) for guitar in guitars)) + 3  # Add 3 for $ , and . characters
```
Comment (dona-reano) [inline prac_06/guitars.py:31]: Good job using longest length to calculate display formatting 👍

File: prac_06/programming_language.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    """Represent information about a programming language."""
+
+    def __init__(self, name="", typing="", reflection="", year=0):
+        """Initialise a ProgrammingLanguage object.
+
+        name: string, reference name for object
+        typing: string, either Static or Dynamic typing
+        reflection:  Boolean, either True or False
+        year: integer, year language first appeared
+        """
```
Comment (dona-reano) [inline prac_06/programming_language.py:17]: Good details 👍

**Review by dona-reano — COMMENTED** [review]
Well done with your work in Prac 06. Just some minor fixes with variable naming and shortening code lines. Great job :)

**Comment by SallySheldon** [comment]
Thanks @dona-reano, appreciate your feedback - good pick-up with the guitar_to_add variable in the client code for using the Guitar class. :-)

File: prac_06/guitar.py (hunk 1 above)
Comment (lindsaymarkward) [reply prac_06/guitar.py:11]: If you're going to set the year like this with the module, don't think of it as a `CONSTANT` but a `variable`.

Comment (SallySheldon) [reply prac_06/guitar.py:11]: Thanks @lindsaymarkward, of course - that makes much more sense!

**Pull request merged**